findbugsTestCases/src/java/sfBugsNew/Feature313jdk8.java

```diff
@@ -0,0 +1,314 @@
+package sfBugsNew;
+
+import java.io.IOException;
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import edu.umd.cs.findbugs.annotations.ExpectWarning;
+import edu.umd.cs.findbugs.annotations.NoWarning;
+import edu.umd.cs.findbugs.annotations.ValueBased;
+
+@SuppressWarnings("javadoc")
+public class Feature313jdk8 {
+
+    /*
+     * TESTS
+     */
+
+    public static class ReferenceComparisonOfValueBasedClass {
+
+        @ExpectWarning("VBC_REF_COMPARISON")
+        public void referenceComparisonOfOptional() {
+            Optional<String> optional = Optional.of("Foo");
+            Object object = Optional.of("Bar");
+
+            boolean same = optional == object;
+            System.out.println("Reference comparison of 'Optional' yields: " + same);
+        }
+
+        // it would be great if this worked, but 'optionals.get(0)' gets recognized as string
```

Comment:
Any ideas whether we can get this to work?

(There is a similar case [further below](https://github.com/findbugsproject/findbugs/pull/46/files#diff-5ced2e172ef0ce58b2a1ffd689b60e7aR261).)